crates/solver/src/settlement_post_processing/optimize_buffer_usage.rs

```diff
@@ -0,0 +1,104 @@
+use super::SettlementSimulating;
+use crate::settlement::Settlement;
+use shared::token_list::TokenList;
+use web3::types::AccessList;
+
+/// If a settlement only trades trusted tokens try to optimize it by trading with internal buffers.
+pub async fn optimize_buffer_usage(
+    settlement: Settlement,
+    access_list: Option<AccessList>,
+    market_makable_token_list: &Option<TokenList>,
+    settlement_simulator: &impl SettlementSimulating,
+) -> Settlement {
+    // We don't want to buy tokens that we don't trust. If no list is set, we settle with external liquidity.
+    if !market_makable_token_list
+        .as_ref()
+        .map(|list| is_only_selling_trusted_tokens(&settlement, list))
+        .unwrap_or(false)
+    {
+        return settlement;
+    }
+
+    let optimized_settlement = settlement.clone().without_onchain_liquidity();
+
+    if settlement_simulator
+        .settlement_would_succeed(optimized_settlement.clone(), access_list)
+        .await
```

Comment:
Not something that was introduced here, but I am just noticing it now - we need to re-compute access lists when we internalize liquidity.

Reply:
Yeah, I stumbled across this as well and something else: #438.
I researched the access lists thing a bit and I think it's actually not a problem. The submitter recomputes the access lists itself [[0](https://github.com/cowprotocol/services/blob/main/crates/solver/src/settlement_submission/submitter.rs#L436-L450)] so not problem there and for the optimization logic the recycled access lists would only cause the settlement to report higher costs but we only check whether the optimized settlement reverts or not.

Reply:
Sounds like its not an issue after all.

Reply:
Since the access lists don't matter I will remove them from the `PostProcessingPipeline` function signatures to avoid confusion in the future.